packages/ai/core/generate-transcript/generate-transcript-result.ts

```diff
@@ -0,0 +1,24 @@
+import { GeneratedTranscript } from '../generate-transcript';
+import { TranscriptionWarning } from '../types/transcription-model';
+import { TranscriptionModelResponseMetadata } from '../types/transcription-model-response-metadata';
+
+/**
+The result of a `generateTranscript` call.
+It contains the transcript and additional information.
+ */
+export interface GenerateTranscriptResult {
+  /**
+  The transcript that was generated.
+   */
+  readonly transcript: GeneratedTranscript;
```

Comment:
wondering about renaming to `text`

Reply:
let's remove 1 level of indirection, i.e. move all props from GeneratedTranscript into the result and remove GeneratedTranscript